tabrepo/nips2025_utils/end_to_end_single.py

```diff
@@ -307,6 +335,94 @@ def to_results(self) -> EndToEndResultsSingle:
             hpo_results=self.hpo_results,
         )
 
+    # FIXME: Refactor
+    @staticmethod
+    def create_and_cache_end_to_end_results(
+        path_raw: str | Path,
+        num_cpus: int | None = None,
+        artifact_name: str | None = None,
+        task_metadata: pd.DataFrame | None = None,
+    ) -> EndToEndResultsSingle:
+        """Create and cache end-to-end results for all methods in the given directory.
+
+        Args:
+            path_raw (str | Path): Path to the directory containing raw results.
+            num_cpus (int | None): Number of CPUs to use for parallel processing.
+                If None, it will use all available CPUs.
+            artifact_name (str | None): Optional name to distinguish different runs of
+                the same method.
+        """
+        if num_cpus is None:
+            num_cpus = len(os.sched_getaffinity(0))
+
+        print("Get results paths...")
+        file_paths = fetch_all_pickles(
+            dir_path=path_raw, suffix="results.pkl"
+        )
```

Comment:
@LennartPurucker `fetch_all_pickles` I believe should be just as fast as `fetch_all_pickles_fast` now. Let me know if that is the case.